test/functorch/test_control_flow.py

```diff
@@ -1136,6 +1137,262 @@ def false_fn(x):
             expected_grads = torch.autograd.grad(fn(x), (x,), grad_out)
             self.assertEqual(expected_grads, grads)
 
+    def _test_cond_autograd(self, cond_fct, pred, true_fn, false_fn, operands):
+        from torch.fx.passes.shape_prop import _extract_tensor_metadata, TensorMetadata
+
+        # This is a helper function that extracts the metadata from the tensor and
+        # sets the requries_grad flag to false. This is needed as we compare the
+        # metadata of the operands and the gradients
+        def _extract_tensor_metadata_except_requires_grad(arg):
+            metadata = _extract_tensor_metadata(arg)
+            metadata = TensorMetadata(
+                metadata.shape,
+                metadata.dtype,
+                False,
+                metadata.stride,
+                metadata.memory_format,
+                metadata.is_quantized,
+                metadata.qparams,
+            )
+            return metadata
+
+        # Comparison of FWD path
+        y_cond = cond_fct(pred(*operands), true_fn, false_fn, operands)
+        operands_forced_grad = [o.clone().detach() for o in operands]
+        for o in operands_forced_grad:
+            o.requires_grad = True
+        y_exp = true_fn(*operands_forced_grad) if pred(*operands_forced_grad) else false_fn(*operands_forced_grad)
+        self.assertEqual(y_cond, y_exp)
+
+        # Comparison of BWD path
+        grad_operands = [o for o in operands if o.requires_grad]
+        grad_exp_operands = [o for o in operands_forced_grad if o.requires_grad]
+        
+        # Check if at least some operators require grads
+        if len(grad_operands) > 0:
+            grad_cond = torch.autograd.grad(y_cond, grad_operands, allow_unused=True, retain_graph=True)
```

Comment:
The naming of the variables here is kind of confusing to me. IIUC, 
- `y_cond` is output of cond, right? so maybe just `cond_outputs`
- `grad_operands` are the inputs of cond that we want to compute graident for, so `cond_inputs`, maybe better?
- `grad_cond` is actually the gradient of those operands which require gradient. So a better name could be 'grad_inputs'?

So the call to torch.autograd.grad would looke like:
```python
grad_inputs = torch.autograd.apply(cond_outputs, cond_inputs, allow_unused=True) 
```
wdyt?

Reply:
I agree, your naming is better-suited. I changed it